### PR TITLE
Allow trying multiple sources

### PR DESCRIPTION
**Pull request by ahstro**
This would let the user supply an array of image strings to `src`, to be tested before falling back on `fallbackImage`.
Backwards compatability with a single string has been kept, so no worries about that :)

### CODE REVIEW

**Comment by conorhastings** [comment]
@ahstro I think it makes more sense to have multiple fallbacks but one primary image. What do you think?

**Comment by ahstro** [comment]
Well, that depends. The definition I use is that the `fallbackImage` is an image the user will know for sure is there, while images supplied to `src` will be ones that the user is unsure of whether they'll be there or not, and if you subscribe to that, it makes more sense to supply multiple `src` images with one `fallbackImage`.

Also, don't merge just yet, I was optimistic about a change I made just before submitting and the code broke. :sweat_smile:

**Comment by conorhastings** [comment]
@ahstro cool sounds good. tag me again when it's ready and i'll think about the fallbackImage/src thing

**Comment by ahstro** [comment]
Okay, that should be it. I'm a bit stressed, so I might've missed something, but it feels like I tested it quite extensively.
Give me a shout if I need to fix something :)

File: src/index.js [hunk 1]
```diff
@@ -25,18 +26,31 @@ class ReactImageFallback extends Component {
 		this.displayImage.onload = null;
 	}
 
+	setDisplayImageSrc(foo, bar, oob) {
+		this.displayImage.src = this.state.imageSource
+	}
+
 	setDisplayImage(image, fallback) {
+		const isArr = Array.isArray(image)
```
Comment (conorhastings) [inline src/index.js]: can we use a semicolon here.

File: src/index.js [hunk 2]
```diff
@@ -25,18 +26,31 @@ class ReactImageFallback extends Component {
...
 			this.setState({
-				imageSource: image
-			});
+				imageSource: isArr
+					? image[this.state.imageIndex] || fallback
+					: image || fallback
+			}, this.setDisplayImageSrc);
 		};
-		this.displayImage.src = image;
+		this.setDisplayImageSrc()
```
Comment (conorhastings) [inline src/index.js]: please use a semicolon here as well.

File: src/index.js [hunk 3]
```diff
@@ -25,18 +26,31 @@ class ReactImageFallback extends Component {
 		this.displayImage.onload = null;
 	}
 
+	setDisplayImageSrc(foo, bar, oob) {
```
Comment (conorhastings) [inline src/index.js]: can we remove the arguments if they're not neccesary

File: src/index.js [hunk 4]
```diff
@@ -25,18 +26,31 @@ class ReactImageFallback extends Component {
...
+		this.displayImage.src = this.state.imageSource
```
Comment (conorhastings) [inline src/index.js]: semicolon her as well please.

**Comment by conorhastings** [comment]
@ahstro can you update the readme as well to reflect these changes?

**Comment by conorhastings** [comment]
after giving this some more thought, i'd really prefer if the array of images was something that was allowed for the fallback instead of the src. Theoretically you could use it in the same sense but to me this just makes a lot more sense, i'd be happy to continue the PR if you feel differently!

**Comment by ahstro** [comment]
Fixed the semicolons and arguments. Sorry about that, I was in quite a rush when I pushed..

I guess, if one can supply an array of images, the need for separate `src` and `fallbackImage` attributes is really quite obsolete (since one could put the fallback image as the last member of the array), and would only be needed for backwards compatibility and syntactic sugar.

If you prefer that the array be supplied as the `fallbackImage` prop, I can try to fix that tomorrow or something (unless you want to do it yourself).

**Comment by conorhastings** [comment]
@ahstro that's a really good point, lets just remove fallback image and allow an array for src. If you can do that, i'll update the documentation and publish a new major version.

**Comment by ahstro** [comment]
Is it really necessary to remove `fallbackImage`? Not doing so would allow you to just publish a minor version and not break any code, and I think it's syntactically better and more user friendly to at least have a dedicated fallback prop available.
It could still work the same underneath, the fallback could just be concated onto the src at initialization.

I'll probably start fixing whatever you decide tomorrow. My CPU fan unfortunately broke the other day and I can't really work on anything right now.

**Comment by conorhastings** [comment]
Sorry to hear that, but no rush!

I hear what you're saying and I am inclined to agree that a dedicated prop makes sense just for ease of entry to the component. 

I still think where we differ is on where we would place the array of images versus single image. I still think it's my preference to have the fallbackImage contain the array, let me know what you think and if you're up for doing that.

**Comment by ahstro** [comment]
So where do we land on this? I'm fine with either just a `src` attribute that takes either an array or a string, or a separate `src` and `fallbackImage` attribute.

The way to do it that makes most sense to me is to just concat `src` and `fallbackImage` so that it, in affect, is a single array. This would technically allow people to put arrays in either `src` or `fallbackImage`. If you want it to force a string in `src`, that's fine too.

**Comment by conorhastings** [comment]
the concat with allowing arrays for either sounds good to me :+1: thanks @ahstro !

**Comment by ahstro** [comment]
Cool, I'll get on it in a bit :)

**Comment by ahstro** [comment]
Semicolons are in place. This should be it (fingers crossed) :)

**Comment by conorhastings** [comment]
sweet, I will take a look as soon as time allows, thanks.

**Comment by ahstro** [comment]
Forgot to check whether or not `initialImage` worked; it didn't.

**Comment by ahstro** [comment]
Okay, I'm not sure what's wrong with my brain this week, but I've apparently completely forgotten how to write code. Don't waste your time looking at this just yet.. I'll do some serious, relaxed, no-rush coding and testing of this myself and won't bother you again until I'm 100% sure it works as intended.
Sorry..

**Comment by conorhastings** [comment]
looks great, tested it out and it works great, going to update some docs then i'll publish a new version

**Comment by conorhastings** [comment]
@ahstro published as `v2.1.0` thanks again!

**Comment by ahstro** [comment]
Wait, hold on, what, no, I'm pretty sure that wasn't working at all.. Lots of infinite loops and weird stuff in some scenarios. I have some code that seems to be working better, I've tested it quite thoroughly, but I'm not guaranteeing anything.

**Comment by conorhastings** [comment]
hmmm...ok...i use it in a very simple app and it seemed to be working but i only tested briefly, i've reverted the PR and released a ptch version for now, i'll try and look tomorrow if you make a new PR.

**Pull request merged**